web/src/main/java/org/springframework/security/web/util/matcher/AntPathRequestMatcher.java

```diff
@@ -159,9 +160,14 @@ public boolean matches(HttpServletRequest request) {
 
 	@Override
 	public MatchResult matcher(HttpServletRequest request) {
-		if (this.matcher == null || !matches(request)) {
+		if (!matches(request)) {
 			return MatchResult.notMatch();
 		}
+
+		if (this.matcher == null) {
```

Comment:
Changing the semantics in a minor release worries me a bit. Why is a null matcher a match now instead of not a match?

Reply:
This matcher is only used to extract variables. If we have a `MATCH_ALL` pattern then `matcher` will be set to `null` and this method returns `MatchResult.notMatch` so I think this is a bug.

Reply:
I see. Is the change necessary for this PR? If not, please create a separate ticket to get the bug addressed. This makes it easier to backport to earlier branches as necessary.

Reply:
Yes, this is necessary, since I use `MatchResult` in `FilterInvocationAuthorizationManager` to get variables. Without this change, tests where the `MATCH_ALL` pattern is used will fail.